src/public_key.rs

```diff
@@ -59,22 +59,14 @@ impl PublicKeyService for PublicKeyGrpc {
         }
 
         let response = datastore::add_public_key(&self.db_pool, request).await?;
-        let key_bytes = match response.public_key.clone().unwrap().key.unwrap() {
-            Key::Secp256k1(data) => data,
+        let key = match response.public_key.clone().unwrap().key.unwrap() {
+            Key::Secp256k1(data) => base64::encode(data),
         };
-        let key = std::str::from_utf8(&key_bytes)
-            .map_err(Into::<OsError>::into)
-            // TODO remove this when we have public_key RPCs using the OsError across the board
-            .map_err(Into::<Status>::into)?
-            .to_owned();
 
+        let mut cache = self.cache.lock().unwrap();
```

Comment:
This has a different scope whether it is inside the if statement or outside like this. When `cache` goes out of scope it will hit unlock. You can explicitly unlock, but it's more idiomatic rust to let the code scope handle it. This syntax is often used though - basically adding an arbitrary scope around the lock call.

```rust
{
        let mut cache = self.cache.lock().unwrap();
        if response.url.is_empty() {
            cache.add_local_public_key(key);
        } else {
            cache.add_remote_public_key(key, response.url.clone());
        }
}
```

Reply:
ah, got it, I'll change it back. It just looked like a duplicated line to me, didn't know about the going out of scope bit, thanks for the insight!